core/src/test/scala/integration/kafka/api/AdminClientIntegrationTest.scala

```diff
@@ -2422,4 +2424,23 @@ object AdminClientIntegrationTest {
 
     assertEquals(Defaults.CompressionType.toString, configs.get(brokerResource).get(KafkaConfig.CompressionTypeProp).value)
   }
+
+  private def getTopicMetadata(client: Admin,
+                               topic: String,
+                               describeOptions: DescribeTopicsOptions = new DescribeTopicsOptions,
+                               expectedNumPartitionsOpt: Option[Int] = None): TopicDescription = {
+    var result: TopicDescription = null
+
+    TestUtils.waitUntilTrue(() => {
+      val topicResult = client.describeTopics(Set(topic).asJava, describeOptions).values.get(topic)
+      if (topicResult.isCompletedExceptionally) {
```

Comment:
I think it would be useful to capture the exception if we time out.

Reply:
I think we need a retry on UnknownTopicOrPartitionException only - I will make the change.